src/Command/EnumCommand.php

```diff
@@ -82,12 +97,104 @@ public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionPar
 
         $parser->setDescription(
             'Bake backed enums for use in models.'
-        )->addOption('int', [
-            'help' => 'Using backed enums with int instead of string as return type',
+        )->addArgument('name', [
+            'help' => 'Name of the enum to bake. You can use Plugin.name to bake plugin enums.',
+            'required' => true,
+        ])->addArgument('cases', [
+            'help' => 'List of either `one,two` for string or `0:foo,1:bar` for int type.',
+        ])->addOption('int', [
+            'help' => 'Using backed enums with int instead of string as return type.',
             'boolean' => true,
             'short' => 'i',
         ]);
 
         return $parser;
     }
+
+    /**
+     * @param string|null $casesString
+     * @return array<int|string, string>
+     */
+    protected function parseCases(?string $casesString, bool $int): array
+    {
+        if ($casesString === null) {
+            return [];
+        }
+
+        $enumCases = explode(',', $casesString);
+
+        $definition = [];
+        foreach ($enumCases as $k => $enumCase) {
+            $key = $value = trim($enumCase);
+            if (str_contains($key, ':')) {
+                $value = trim(mb_substr($key, strpos($key, ':') + 1));
+                $key = mb_substr($key, 0, strpos($key, ':'));
+            } elseif ($int) {
+                $key = $k;
+            }
+
+            $definition[$key] = $value;
+        }
+
+        return $definition;
+    }
+
+    /**
+     * @param array<int|string, string> $definition
+     * @return bool
+     */
+    protected function isOfTypeInt(array $definition): bool
+    {
+        if (!$definition) {
+            return false;
+        }
+
+        foreach ($definition as $key => $value) {
+            if (!is_int($key)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array<int|string, string> $cases
+     * @return array<string>
+     */
+    protected function formatCases(array $cases): array
+    {
+        $formatted = [];
+        foreach ($cases as $case => $alias) {
+            $alias = mb_strtoupper(Inflector::underscore($alias));
+            if (is_string($case)) {
+                $case = '\'' . $case . '\'';
```

Comment:
Do we need to escape any `'` or raise an error should the `case` contain quotes or other characters not allowed in PHP constant names?

Reply:
According to PHP constants are

    ^[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*$

but is the same true for enum cases though?

It would still be wise to limit our bake to those IMO, as the other characters seem pointless here to support.